src/formatDate.js

```diff
@@ -8,7 +8,49 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const separator = toFormat[3];
+
+  const dateArr = date.split(oldSeparator);
+
+  let yearIndex;
+
+  if (fromFormat.indexOf('YYYY') !== -1) {
+    yearIndex = fromFormat.indexOf('YYYY');
+  } else {
+    yearIndex = fromFormat.indexOf('YY');
+  }
+
+  const monthIndex = fromFormat.indexOf('MM');
+  const dayIndex = fromFormat.indexOf('DD');
+
+  let year = dateArr[yearIndex];
+  const month = dateArr[monthIndex];
+  const day = dateArr[dayIndex];
+
+  if (year.length === 2 && toFormat.includes('YYYY')) {
+    if (year < 30) {
+      year = `20${year}`;
+    } else {
+      year = `19${year}`;
+    }
+  } else if (year.length === 4 && toFormat.includes('YY')) {
+    year = year.slice(-2);
+  }
+
+  const dateParts = {
+    YYYY: year,
+    YY: year,
+    MM: month,
+    DD: day,
+  };
+
+  const result = toFormat
+    .filter((part) => part !== separator)
+    .map((part) => dateParts[part])
+    .join(separator);
```

Comment:
The `filter` method is used to remove the separator from the `toFormat` array. However, this assumes that the separator is part of the array, which may not be the case if the separator is not correctly identified. Ensure that the separator handling logic is robust and correctly identifies and removes separators from the format arrays.